File: src/main/java/com/sunshines/bookstore/Service/UserDetailsSvc.java
package com.sunshines.bookstore.Service;

import com.sunshines.bookstore.Model.User;
import com.sunshines.bookstore.Repository.UserRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class UserDetailsSvc implements UserDetailsService {
    private UserRepository userRepository;

    public UserDetailsSvc(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public User loadUserByUsername(String username) throws UsernameNotFoundException {
        User user = userRepository.findFirstByEmail(username);
//        System.out.print(user.getRole().getAuthority());
        if (user == null) {
            throw new UsernameNotFoundException(username);
        }
        return user;
    }

    public UserDetails loadUserById(int userId) {
        User user = userRepository.findFirstById(userId);
        if (user == null) {
            throw new UsernameNotFoundException(Integer.toString(userId));
        }
        return user;

    }
}

File: src/main/java/com/sunshines/classrooms/Service/UserDetailsSvc.kt
package com.sunshines.classrooms.Service

import com.sunshines.classrooms.Model.User
import com.sunshines.classrooms.Repository.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsSvc : UserDetailsService {

    @Autowired
    lateinit var userRepository: UserRepository

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): User {
//        System.out.print(user.getRole().getAuthority());
        return userRepository.findFirstByEmail(username) ?: throw UsernameNotFoundException(username)
    }

    fun loadUserById(userId: Int): UserDetails {
        return userRepository.findFirstById(userId) ?: throw UsernameNotFoundException(Integer.toString(userId))

    }
}